pkg/apis/seedmanagement/validation/managedseed.go

```diff
@@ -0,0 +1,327 @@
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("seedSelector"), "seed selector is forbidden"))
+	}
+
+	// Validate gardenlet config
+	allErrs = append(allErrs, configvalidation.ValidateGardenletConfiguration(gardenletConfig, fldPath)...)
+
+	if gardenletConfig.SeedConfig != nil {
+		seedConfigPath := fldPath.Child("seedConfig")
+
+		// Convert seed to internal version
+		seed, err := helper.ConvertSeed(&gardenletConfig.SeedConfig.Seed)
+		if err != nil {
+			allErrs = append(allErrs, field.InternalError(seedConfigPath, fmt.Errorf("could not convert seed config: %v", err)))
+			return allErrs
+		}
+
+		// Validate seed
+		allErrs = append(allErrs, validateSeed(seed, seedConfigPath)...)
+	}
+
+	if gardenletConfig.GardenClientConnection != nil {
+		allErrs = append(allErrs, validateGardenClientConnection(gardenletConfig.GardenClientConnection, bootstrap, mergeWithParent, fldPath.Child("gardenClientConnection"))...)
+	}
+
+	return allErrs
+}
+
+func validateGardenletConfigurationUpdate(newGardenletConfig, oldGardenletConfig *config.GardenletConfiguration, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if newGardenletConfig.SeedConfig != nil && oldGardenletConfig != nil {
+		seedConfigPath := fldPath.Child("seedConfig")
+
+		// Convert new seed to internal version
+		newSeed, err := helper.ConvertSeed(&newGardenletConfig.SeedConfig.Seed)
+		if err != nil {
+			allErrs = append(allErrs, field.InternalError(seedConfigPath, fmt.Errorf("could not convert seed config: %v", err)))
+			return allErrs
+		}
+
+		// Convert old seed to internal version
+		oldSeed, err := helper.ConvertSeed(&oldGardenletConfig.SeedConfig.Seed)
+		if err != nil {
+			allErrs = append(allErrs, field.InternalError(seedConfigPath, fmt.Errorf("could not convert seed config: %v", err)))
+			return allErrs
+		}
+
+		// Validate seed
+		allErrs = append(allErrs, validateSeedUpdate(newSeed, oldSeed, seedConfigPath)...)
+	}
+
+	return allErrs
+}
+
+func validateSeed(seed *gardencore.Seed, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// Ensure name is not specified since it will be set by the controller
+	if seed.Name != "" {
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("metadata", "name"), "seed name is forbidden"))
+	}
+
+	// Validate labels, annotations, and spec
+	allErrs = append(allErrs, metav1validation.ValidateLabels(seed.Labels, fldPath.Child("metadata", "labels"))...)
+	allErrs = append(allErrs, apivalidation.ValidateAnnotations(seed.Annotations, fldPath.Child("metadata", "annotations"))...)
+	allErrs = append(allErrs, corevalidation.ValidateSeedSpec(&seed.Spec, fldPath.Child("spec"))...)
+
+	return allErrs
+}
+
+func validateSeedUpdate(newSeed, oldSeed *gardencore.Seed, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	allErrs = append(allErrs, corevalidation.ValidateSeedSpecUpdate(&newSeed.Spec, &oldSeed.Spec, fldPath.Child("spec"))...)
```

Comment:
Should we also call `ValidateObjectMetaUpdate` here?

Reply:
I don't think so. Looking at the code at https://github.com/kubernetes/apimachinery/blob/ee96543cc6730a24af1c7323dac47e52c340ba39/pkg/api/validation/objectmeta.go#L233, it mostly validates updates to fields that are ignored (they shouldn't be there at all, see previous comments). The only exception are `Name` (immutable), `Labels`, and `Annotations`, but these are validated by `ValidateManagedSeed` that is called by `ValidateManagedSeedUpdate`. As `Name` is forbidden checking for immutable is meaningless.

Reply:
Another consideration: if we call `ValidateObjectMetaUpdate` here, we should also call `ValidateObjectMeta` in `validateSeed`, otherwise it would be strange - but this is not working, since it would treat e.g. `Name` as required (and in our case, it's forbidden).

Reply:
Ok, got the motivation now, makes sense. Thanks for clarifying.